java-sdk/src/main/java/com/adobe/target/edge/client/ondevice/DefaultRuleLoader.java

```diff
@@ -63,12 +63,15 @@ public synchronized void start(final ClientConfig clientConfig) {
         if (started) {
             return;
         }
-        ObjectMapper mapper = new JacksonObjectMapper();
+
         byte[] artifactPayload = clientConfig.getOnDeviceArtifactPayload();
+
         if (artifactPayload != null) {
-            String payload = new String(artifactPayload, StandardCharsets.UTF_8);
-            OnDeviceDecisioningRuleSet ruleSet = mapper.readValue(payload, new GenericType<OnDeviceDecisioningRuleSet>() {});
+            ArtifactObfuscator artifactObfuscator = new ArtifactObfuscator();
+            String payload = artifactObfuscator.unobfuscate(clientConfig.getOrganizationId(), artifactPayload);
```

Comment:
This call could throw an exception which won't be properly caught, causing the whole start method to fail. This call, or perhaps this whole artifactPayload block, should be protected by a try/catch to ensure the whole method doesn't fail if the user passes in an invalid payload.

Reply:
The exception should surface all the way to the top and ultimately fail creating the Target client with an exception.

Or if we protect it here with a try block and just do a `logger.error()`, how is customer being notified?

Reply:
There is a callback to get exceptions and this exception should be passed in there if it is encountered. This is only used for people who want to immediately use on device decisioning and if they pass in an invalid artifact (which is pretty likely when they migrate to a new version of the client which supports a new version of the artifact, but they are passing in a cached version of the old artifact) then the worst case it that they have to wait a bit longer for the new artifact to download. This is preferable to all of their clients throwing exceptions in this case since the user doesn't really know anything about artifact versioning and whether or not their old artifact is compatible with the current version of the SDK.

Reply:
Fixed.